crypto/CMakeLists.txt

```diff
@@ -455,13 +455,22 @@ add_library(
 )
 
 if(FIPS_SHARED)
-  # Rewrite libcrypto.so to inject the correct module hash value. This assumes
-  # UNIX-style library naming, but we only support FIPS mode on Linux anyway.
+  # Rewrite libcrypto.so (or libcrypto.dylib) to inject the correct module
+  # hash value. For now we support the FIPS build only on Linux and macOS.
+
+  if (APPLE)
+    set(SHARED_LIB_NAME "libcrypto.dylib")
+    set(INJECT_HASH_MACOS_FLAG "-macos")
+  else()
+    set(SHARED_LIB_NAME "libcrypto.so")
+    set(INJECT_HASH_MACOS_FLAG "")
+  endif()
```

Comment:
Can we get rid of all of this and use generator expressions?

Reply:
got rid of `SHARED_LIB_NAME`, `INJECT_HASH_MACOS_FLAG` will be handled in a future PR as discussed offline